### ISSUE TITLE
Adapt interface to new eth style derivation

### ISSUE THREAD

**Issue by joelamouche** [polkadot-js/apps#4722]
- make `m/44'/60'/0'/0/0` the default "secret derivation path" when deriving from mnemonic
- verify that derivation path is m/number(') etc

**Comment by joelamouche** [comment]
I'll do it

**Comment by joelamouche** [comment]
@jacogr Do you mind republishing https://www.npmjs.com/package/@polkadot/ui-keyring ? It doesn't seem like it was published again and this is the package used in the app.
Thanks in advance

**Comment by jacogr** [comment]
I am in the process of going wasm -> common -> api -> ui -> apps. It takes a while, at step 2 atm :)

**Comment by joelamouche** [comment]
Great thanks 🙏

**Comment by joelamouche** [comment]
I also want to set “slab nerve salon plastic filter inherit valve ozone crash thumb quality whale” as the default mnemonic for ethereum addresses, which is derived from seed "ethereum".
I propose this becomes also the default set of dev addresses loaded by the api in dev mode

**Comment by jacogr** [comment]
The seed is not a Substrate-standard, so unlikely to go in.

**Comment by jacogr** [comment]
Final step at https://github.com/polkadot-js/apps/pull/4723 (UI doesn't need to be bumped, the apps UI uses resolutions, so will use the latest util/util-crypto/keyring)

**Comment by joelamouche** [comment]
Ok so trying to understand your answer "The seed is not a Substrate-standard, so unlikely to go in." and realised there are still some things I dont understand:
- createTestKeyring in common: is that where the dev addresses are declared? So as is, Gerald will be loaded?
- In this function, the seed is the sr25519 seed?
- are you saying that changing the default mnemonic is fine but I will have to declare the dev addresses as private/public key pair?
- what does it have to do with substrate?

**Comment by jacogr** [comment]
Gerald won’t be loaded. The test keyring only used sr25519. (I did comment on this)

True seed is standard in Substrate for dev accounts. In subkey when you just input „//Alice“ (no mnemonic) it uses the standard Substrate mnemonic for dev seeds. This is what the keyring emulates.

**Comment by joelamouche** [comment]
@jacogr 

So I'm trying to debug the app and I'm getting  "Invalid publicKey provided" when I create the key
which is coming from 
```

  saveAccount(pair, password) {
    this.addTimestamp(pair);
    const json = pair.toJson(password);
    this.keyring.addFromJson(json);
    this.accounts.add(this._store, pair.address, json, pair.type);
    return json;
  }
```
from node_modules/@polkadot/ui-keyring/Keyring.js

I created this branch in common to reproduce the test: https://github.com/PureStake/common/tree/jlm-fix-eth-derivation

What Im confused about is that toJson doesn't take the secretkey into account so I dont understand how saveAccount gets the good secret key.

PS: Could we also not update common dependancy automatically? My whole team is blocked now because derivation is broken in the interface

**Comment by joelamouche** [comment]
Here is my progress on the app update btw: https://github.com/polkadot-js/apps/pull/4732

**Comment by jacogr** [comment]
https://github.com/polkadot-js/common/blob/master/packages/keyring/src/pair/decode.ts#L37

TL;DR it assumed 32 bytes, but since it is the last in the structure, the explicit length can be removed.

No, we cannot auto-update deps. In most cases they require updates.

**Comment by joelamouche** [comment]
>the explicit length can be removed

What do you mean by this?

>No, we cannot auto-update deps. 

Ha ok good. Next time please let me update the dependancy

**Comment by joelamouche** [comment]
Here are my tests showing where the error is:https://github.com/PureStake/common/commit/6d4ca1f715a3732cc9cef2bf4292009830457d7f
I just don't know how to fix it

**Comment by jacogr** [comment]
When it decodes, it does this -

`const publicKey = encoded.subarray(pubOffset, pubOffset + PUB_LENGTH);` (that is the line linked above)

So it always contrains publicKeys to 32 bytes.

**Comment by joelamouche** [comment]
Sorry I still don't understand what change you are suggesting. Maybe it would be easier if you just branched out of my branch?
In the createFromJson you pasted above there are two things that I don't understand:
- createPair is called with an empty secretKey, so how is it supposed to recreate the pair that was turned into Json
- publicKey, when logged, is 20bytes long because it is copied from address, which is 20bytes, and that's where by error comes form when calling createPair

**Comment by jacogr** [comment]
Exactly. The `address` in the Ethereum pair should not be the address.

It should be the full publicKey, hex-encode.

**Comment by joelamouche** [comment]
OK makes sens thank you! Im going to reverse that change. But not for ecdsa, because it seems doing so breaks the ecdsa tests
